src/test/java/pl/task/parkingmeter/repository/VehicleRepositoryTest.java

```diff
@@ -0,0 +1,97 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+import org.springframework.test.context.junit4.SpringRunner;
+import pl.task.parkingmeter.entity.Vehicle;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+public class VehicleRepositoryTest {
+
+    private final String REG_NUMBER_FOR_REGULAR = "QWERTY";
+    private final String REG_NUMBER_FOR_PAID_EARLIER = "ASDFG";
+    private final String REG_NUMBER_FOR_DISABLED = "ZXCVB";
+
+    @Autowired
+    private TestEntityManager entityManager;
+    @Autowired
+    private VehicleRepository vehicleRepository;
+
+    @Before
+    public void setup() {
+        Vehicle vehicleRegular = new Vehicle();
+        vehicleRegular.setRegNumber(REG_NUMBER_FOR_REGULAR);
+        vehicleRegular.setOwnerDisabled(false);
+        vehicleRegular.setCurrency("PLN");
+
+        LocalDateTime localDateTime = LocalDateTime.of(2018, 8, 20, 7, 0, 0);
+        vehicleRegular.setPayDate(localDateTime);
+        entityManager.persist(vehicleRegular);
+
+        Vehicle vehiclePaidEarlier = new Vehicle();
+        vehiclePaidEarlier.setRegNumber(REG_NUMBER_FOR_PAID_EARLIER);
+        vehiclePaidEarlier.setPaid(true);
+        vehiclePaidEarlier.setCurrency("PLN");
+        localDateTime = LocalDateTime.of(2018, 8, 21, 7, 0, 0);
+        vehiclePaidEarlier.setPayDate(localDateTime);
+        entityManager.persist(vehiclePaidEarlier);
+
+        Vehicle vehicleWithDisabledOwner = new Vehicle();
+        vehicleWithDisabledOwner.setRegNumber(REG_NUMBER_FOR_DISABLED);
+        vehicleWithDisabledOwner.setPaid(false);
+        vehicleWithDisabledOwner.setCurrency("PLN");
+        vehicleWithDisabledOwner.setOwnerDisabled(true);
+        localDateTime = LocalDateTime.of(2018, 8, 22, 7, 0, 0);
+        vehicleWithDisabledOwner.setPayDate(localDateTime);
+        entityManager.persist(vehicleWithDisabledOwner);
+
+    }
+
+    @Test
+    public void findVehiclesByIsPaidFalse() {
+        List<Vehicle> result = vehicleRepository.findVehiclesByPaidFalse();
+        assertEquals(2, result.size());
+        assertEquals(REG_NUMBER_FOR_REGULAR, result.get(0).getRegNumber());
+        assertEquals(REG_NUMBER_FOR_DISABLED, result.get(1).getRegNumber());
+    }
+
+    @Test
+    public void findVehicleByRegNumberAndIsPaidFalse() {
+
+        Optional<Vehicle> resultForRegular = vehicleRepository.findVehicleByRegNumberAndPaidFalse(REG_NUMBER_FOR_REGULAR);
+        assertEquals(REG_NUMBER_FOR_REGULAR, resultForRegular.get().getRegNumber());
+
+        Optional<Vehicle> resultForPaidEarlier = vehicleRepository.findVehicleByRegNumberAndPaidFalse(REG_NUMBER_FOR_PAID_EARLIER);
```

Comment:
to są 2 osobne testy - w wielu innych miejscach masz ten sam problem

Reply:
przerobione